pkg/cloudprovider/aws/capacity.go

```diff
@@ -35,34 +49,57 @@ type Capacity struct {
 	instanceTypeProvider   *InstanceTypeProvider
 }
 
+// AWSConstraints are AWS specific constraints
+type AWSConstraints struct {
+	cloudprovider.Constraints
+	CapacityType string
+}
+
+func NewAWSConstraints(constraints *cloudprovider.Constraints) *AWSConstraints {
+	capacityType, ok := constraints.Labels[capacityTypeLabel]
+	if !ok {
+		capacityType = capacityTypeOnDemand
+	}
+	return &AWSConstraints{
+		Constraints:  *constraints,
+		CapacityType: capacityType,
+	}
+}
+
 // Create a set of nodes given the constraints.
 func (c *Capacity) Create(ctx context.Context, constraints *cloudprovider.Constraints) ([]cloudprovider.Packing, error) {
-	// 1. Retrieve normalized zones from constraints or all zones that the cluster spans if no zonal constraints are specified
-	zonalSubnetOptions, err := c.vpcProvider.GetZonalSubnets(ctx, constraints, c.spec.Cluster.Name)
+	// 1. Create AWS Cloud Provider constraints and apply defaults
+	awsConstraints := NewAWSConstraints(constraints)
```

Comment:
consider calling the other one cloudprovider constraints

Reply:
good call 👍